### ISSUE TITLE
Question about Gumbel Sampling

### ISSUE THREAD

**Issue by ayumiymk** [yuleiniu/rva#5]
Hi,

I find the distribution, i.e. pi in Eq. (19) and Eq. (20) should use log before added to g. However, I have checked your codes, I find you do not use log on pi. Does it matter?

**Comment by yuleiniu** [comment]
We encourage readers to follow [1] for details of the Gumbel-Max trick. Actually, $\pi$ in our paper represents $\alpha$ in Section 3 rather than the logits.

References
[1] Chris J Maddison, Andriy Mnih, and Yee Whye Teh. The concrete distribution: A continuous relaxation of discrete random variables. arXiv preprint arXiv:1611.00712, 2016.

**Comment by gicheonkang** [comment]
Hi, thank you for sharing you code.
In your paper, You wrote that rva model obtain one-hot vector using Eq. 19. Strictly speaking, it is not correct. 

Rva model obtain one-hot vector one_hot(argmax(Eq. 20)) not Eq. 19.
In other words, softmax operation is always used regardless of the argument (soft / hard). 

For more details, please refer to the F.gumbel_softmax code as below:
https://pytorch.org/docs/stable/_modules/torch/nn/functional.html
